src/prefect/engine/results/constant_result.py

```diff
@@ -0,0 +1,48 @@
+from typing import Any, Optional
+
+from prefect.engine.result import Result
+
+
+class ConstantResult(Result):
+    """
+    Hook for storing and retrieving constant Python objects. Only intended to be used
+    internally.
+
+    Args:
+        - value (Any): the underlying value that we wish to "handle"
+    """
+
+    def __init__(self, value: Any) -> None:
+        self.value = value
+        super().__init__(value=value)
```

Comment:
We have some decisions happening in-flight over in another result handler migration PR, https://github.com/PrefectHQ/prefect/pull/2161/files. One confusing thing about merging this now is that in that other PR we are updating the Result interface to make `value` optional (I should have included/considered it in the interface ticket you worked on, but we realized afterwards about it since we expect users to instantiate Result subclasses as configuration before the value is set, unlike now).

I'm not sure if I just want to wait for #2161 to land in master before updating this, to avoid duplicating changes in the Result base class signature in both PRs, but if I had to anticipate the changes, I'd suggest:

```suggestion
    def __init__(self, value: Any = None, **kwargs) -> None:
        self.value = value
        super().__init__(value=value, **kwargs)
```
This is inspired from https://github.com/PrefectHQ/prefect/pull/2161/files#diff-c94b1139de5da7d4108afdf61930892cR31-R41.

I don't actually see either `value` not being sent or meaningful `**kwargs` to ever actually be used by ConstantResult based on how it is used in the pipeline now, but the former will be part of the base class signature eventually (at least after #2126) and the `**kwargs` thing just feels nice while we're in here, for a use case I may not be considering if a user ever did go rogue and use ConstantResult themselves.

Reply:
Definitely okay with adding the **kwargs! Nothing wrong with future proofing!

Something of this nature sounds fine with me, but I have a couple questions that might come out of not fully understanding its use. Is the constant result handler (and now constant result) _reserved_ for internal use? Or just a concept that happens to be used internally?

The only reason I ask is that the `Constant` versions of the results completely ignores `read` method for updating its internal value, so the only option we have to ever update a value on an instantiated `Constant` is to directly update the `value` like so:

```
constant_result = ConstantResult("my text')
constant_result.value = "my other text"
```

Do you know if there's a need to be able to update a `Constant`'s value after its already been instantiated? Instead of just creating new instances?

Reply:
For posterity from IRL conversation: ConstantResult should not be used externally, it kind of got backed into the Result API from the result handlers API from when we took it out of the Task API. Directly to the question, no we do not want to support updating a `Constant`s value after its already been instantiated, a new instance would need to be made, and we expect the pipeline to handle that.

Reply:
Hey @alexisprince1994, the PR we were waiting for is in master, and we are back on PIN 16 internally so I merged in the latest here. I think the tests will now complain about the result interface for us :) I can add in the changes to the signature later today unless you're already hanging around!